Dockerfile

```diff
@@ -1,38 +1,64 @@
-FROM golang:1.17.3-buster AS build
-ENV GO111MODULE on
+ARG GO_VERSION=1.22
 
-RUN apt update && \
-    apt -y upgrade && \
-    apt -y install clang libltdl-dev && \
-    apt-get clean
+#------------------------------------------------------------------------------
+# Base Debian Image
+#------------------------------------------------------------------------------
+FROM debian:bookworm as base
+ARG GO_VERSION
 
-ADD . $GOPATH/src/github.com/mozilla-services/autograph-edge
+ENV DEBIAN_FRONTEND='noninteractive' \
+    PATH="${PATH}:/usr/lib/go-${GO_VERSION}/bin:/go/bin" \
+    GOPATH='/go'
 
-RUN cd $GOPATH/src/github.com/mozilla-services/autograph-edge && \
-    make install
+## Enable bookworm-backports
+RUN echo "deb http://deb.debian.org/debian/ bookworm-backports main" > /etc/apt/sources.list.d/bookworm-backports.list
+RUN echo "deb-src http://deb.debian.org/debian/ bookworm-backports main" >> /etc/apt/sources.list.d/bookworm-backports.list
 
-RUN apt-get -y remove clang && \
+RUN apt-get update && \
+    apt-get -y upgrade && \
+    apt-get -y install --no-install-recommends \
+        clang \
+        gcc \
+        libltdl-dev \
+        golang-${GO_VERSION} \
+        curl \
+        ca-certificates && \
+    # Cleanup inline with installation to avoid this layer being bloated with
+    # deb packages and other cached data.
     apt-get clean && \
-    apt-get -y autoremove
+    rm -rf /var/lib/apt/lists/*
 
-FROM debian:buster-slim
-EXPOSE 8080
+#------------------------------------------------------------------------------
+# Build Stage
+#------------------------------------------------------------------------------
+FROM base as builder
+ENV GO111MODULE on
+ENV CGO_ENABLED 1
 
-RUN apt update && \
-    apt -y upgrade && \
-    apt -y install libltdl-dev ca-certificates && \
-    apt-get clean
+ADD . /app/src/autograph
 
-COPY --from=build /go/bin/autograph-edge /usr/local/bin
+RUN cd /app/src/autograph && go install .
+
+#------------------------------------------------------------------------------
+# Deployment Stage
+#------------------------------------------------------------------------------
+FROM base
+EXPOSE 8080
 
-RUN addgroup --gid 10001 app && \
-    adduser --gid 10001 --uid 10001 \
-    --home /app --shell /sbin/nologin \
-    --disabled-password app
+# fetch the RDS CA bundles
```

Comment:
Are the RDS certificates for `autograph-edge`? This seems unnecessary.

Reply:
Whoops - this was overly aggressive copy/paste.